src/Components/Components/src/Forms/FieldIdentifier.cs

```diff
@@ -0,0 +1,111 @@
+
+        /// <summary>
+        /// Gets the name of the editable field.
+        /// </summary>
+        public string FieldName { get; }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+            => (Model, FieldName).GetHashCode();
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+            => obj is FieldIdentifier otherIdentifier
+            && otherIdentifier.Model == Model
+            && string.Equals(otherIdentifier.FieldName, FieldName, StringComparison.Ordinal);
+
+        private static void ParseAccessor<T>(Expression<Func<T>> accessor, out object model, out string fieldName)
+        {
+            var accessorBody = accessor.Body;
+
+            // Unwrap casts to object
+            if (accessorBody is UnaryExpression unaryExpression
+                && unaryExpression.NodeType == ExpressionType.Convert
+                && unaryExpression.Type == typeof(object))
+            {
+                accessorBody = unaryExpression.Operand;
+            }
+
+            if (!(accessorBody is MemberExpression memberExpression))
+            {
+                throw new ArgumentException("The accessor is not supported because its body is not a MemberExpression");
+            }
+
+            // Identify the field name. We don't mind whether it's a property or field, or even something else.
+            fieldName = memberExpression.Member.Name;
+
+            // Get a reference to the model object
+            // i.e., given an value like "(something).MemberName", determine the runtime value of "(something)",
+            switch (memberExpression.Expression)
+            {
+                case ConstantExpression constantExpression:
+                    model = constantExpression.Value;
+                    break;
+                case MemberExpression nestedMemberExpression:
+                    // It would be great to cache this somehow, but it's unclear there's a reasonable way to do
+                    // so, given that it embeds captured values such as "this". We could consider special-casing
+                    // for "() => something.Member" and building a cache keyed by "something.GetType()" with values
+                    // of type Func<object, object> so we can cheaply map from "something" to "something.Member".
+                    var modelLambda = Expression.Lambda(nestedMemberExpression);
+                    var modelLambdaCompiled = (Func<object>)modelLambda.Compile();
+                    model = modelLambdaCompiled();
+                    break;
+                default:
+                    // An error message that might help us work out what extra expression types need to be supported
+                    throw new InvalidOperationException($"The accessor is not supported because the model value cannot be parsed from it. Expression: '{memberExpression.Expression}', type: '{memberExpression.Expression.GetType().FullName}'");
```

Comment:
The other thing that's going to come up immediately is indexers. Other than that, I'm not sure MVC supports much else.

Reply:
Indexers aren't supported because they we can't identify them as a field name (we'd need to encode the indexer key in the field name somehow). Maybe one day we might add some extension along those lines, but for now I'd rather not try to solve that.

Reply:
I think we should be concerned about this in the sense that we need to provide guidance for it.

Imagine you're building a form where users can enter multiple emergency contacts, each one with a name and phone number. How would we tell someone to do this?

Reply:
This should be supported (haven't explicitly verified, but will do):

```
@(() => something[someindex].SomeProperty)
```

That will create a FieldIdentifier with model=`something[someindex]` and fieldname=`SomeProperty`.

What's not supported is this:

```
@(() => something[someindex])
```

... and can't be within this model, because there's no value we can give to "fieldname".

Do you think this limitation is reasonable?

> Imagine you're building a form where users can enter multiple emergency contacts, each one with a name and phone number. How would we tell someone to do this?

In the natural way:

```
@foreach (var contact in person.Contacts)
{
    <InputText bind="contact.Phone" /> (etc)
}
```

or even, if your model requires it:

```
@foreach (var contactKey in person.Contacts.Keys)
{
    <InputText bind="person.Contacts[contactKey].Phone" /> (etc)
}
```

Reply:
> Do you think this limitation is reasonable?

Sure, I think that's eminently reasonable. The one place that users will feel sad about it is if if they try to define a `List<string>` or similar